main.py

```diff
@@ -455,17 +457,38 @@ def check_alien_flows(self, switch, verdict_dt: Optional[datetime] = None):
                 self._install_flows(command, flow_dict, [switch], save=False)
                 flows_to_log(
                     log.info,
-                    f"Flows forwarded to switch {dpid} to be deleted. ",
+                    f"Flows forwarded  to be deleted to switch ",
+                    [dpid],
                     flow_dict,
                 )
             except SwitchNotConnectedError:
                 flows_to_log(
                     log.error,
-                    f"Failed to forward flows to switch {dpid} to be deleted. ",
+                    f"Failed to forward flows  to be deleted to switch ",
+                    [dpid],
                     flow_dict,
                 )
 
-    def delete_matched_flows(self, flow_dicts, switches: dict) -> None:
+    def get_flows_by_cookie_ranges(self, dpids: list[str], cookie_ranges_dict: dict[str, list[tuple[int, int]]]) -> dict:
+        """Avoid multiple DB searches when all or consecutive switches have the same cookie range."""
```

Comment:
The original optimization was a bit easier since we had a guarantee that the same ranges of cookies being searched could belong to any of the dpids. Now, you have a case where different dpis might have different cookie ranges, you can still do it in a single query:

1) Naively - just overfetch for the given dpids and the cookie ranges and let `match_flow` not match it later discarding in the runtime. This can become a problem if it overfetches too much hundreds of thousands of flows.
2) More optimized and relying on MongoDB to optimize the query with logic operators: refactor `get_flows_by_cookie_ranges` query to take `cookie_ranges_by_dpid: dict[list[tuple[int, int]], str]`, then you group upfront by cookie ranges and the value would be the dpid, and then perform a query with `$or` and `$and` query operators where each or would be the dpid with each range cookie gte/lte ors and the `$and` with the given dpid. I don't know how MongoDB would perform, but `switch` and `flow.cookie` are indexed so the DB query engine should perform well in a query too.

Reply:
Now matches are added by switch which works for all scenarios that we have. I did some testing as well.
For 1000 EVCs, 10022 Flows, 6 different switches
Search was done between 3 switches `00:01`, `00:02`, `00:03` and all of them had 10 pairs of cookie and mask.
```
cookie_ranges = [
        build_cookie_range_tuple(12249891381106975044, 18446744073709551615),
        build_cookie_range_tuple(12249914558173240905, 18446744073709551615),
        build_cookie_range_tuple(12249941051479127886, 18446744073709551615),
        build_cookie_range_tuple(12249972146531185219, 18446744073709551615),
        build_cookie_range_tuple(12250334800926168900, 18446744073709551615),
        build_cookie_range_tuple(12250348191460937295, 18446744073709551615),
        build_cookie_range_tuple(12250429636197588295, 18446744073709551615),
        build_cookie_range_tuple(12250445488036383816, 18446744073709551615),
        build_cookie_range_tuple(12249982483863610688, 18446744073709551615),
        build_cookie_range_tuple(12250388561362778176, 18446744073709551615),
]
```

Current implementation

- 0.0059986114501953125
- 0.005565643310546875
- 0.0037260055541992188
- 0.003719329833984375
- 0.003556966781616211
- 0.003459453582763672
- 0.0034170150756835938
- 0.0033202171325683594
- 0.0034608840942382812
- 0.003365755081176758

Mean = 0.003958988189697266 seconds


New implementation

- 0.005450010299682617
- 0.004034757614135742
- 0.003678560256958008
- 0.0035314559936523438
- 0.003561258316040039
- 0.0036687850952148438
- 0.003549337387084961
- 0.003516674041748047
- 0.003585338592529297
- 0.0036420822143554688

Mean = 0.0038218259811401366 seconds

Looks like both have the same performance. Let me know if different tests are required.

Reply:
Excellent. 10k flows and with these measured results and similar results are great. 

Did you also have the chance do also check out with the APM? since it's also consolidate 90th percentile it'd be cool to also see there, although based on what you posted here I don't expect surprises. Finally, I recommend you also setting `db.setProfilingLevel(2)`, and then checking if the indexed got hit properly (it should be based on the numbers you're showing here, but good to confirm too)

Reply:
The indexing is `IXSCAN { switch: 1, flow.cookie: 1, state: 1, flow.priority: -1, inserted_at: 1, updated_at: 1 }`
With 1000 EVCs and 10022 flows, I deleted 10 random EVCs, these are the results:
From `flow_manager` old DELETE API request but new implementation for DB lookup
![Screenshot_20241017_083016](https://github.com/user-attachments/assets/24f4feb8-88f4-4a18-9813-8d139ed32052)

From `mef_eline`, this is the 95 percentile sample:
![Screenshot_20241017_083152](https://github.com/user-attachments/assets/35deaa89-ea2f-4e2b-b52e-6f95b3a4d297)